src/H5Spoint.c

```diff
@@ -576,8 +576,14 @@ H5S__point_add(H5S_t *space, H5S_seloper_t op, size_t num_elem, const hsize_t *c
     for (u = 0; u < num_elem; u++) {
         unsigned dim; /* Counter for dimensions */
 
+        /* The following allocation relies on the size of an hcoords_t being
+         * the same as an 'H5S_pnt_node_t *', so fail now if that's not true
+         */
+        HDcompile_assert(sizeof(hcoords_t) >= sizeof(H5S_pnt_node_t *));
+
         /* Allocate space for the new node */
-        if (NULL == (new_node = (H5S_pnt_node_t *)H5FL_ARR_MALLOC(hcoords_t, space->extent.rank)))
+        /* Note: allocating "rank + 1" to allow for 'next' pointer */
+        if (NULL == (new_node = (H5S_pnt_node_t *)H5FL_ARR_MALLOC(hcoords_t, space->extent.rank + 1)))
```

Comment:
How is this allocation not causing constant memory errors in the current version of the library due to being too small?